src/linter.coffee

```diff
@@ -1,38 +1,44 @@
-_ = require 'lodash'
-async = require 'async'
-asyncHandlers = require 'async-handlers'
-fsExtra = require 'fs-extra'
-DependencyLinter = require './linter/dependency_linter'
-UsedModuleFinder = require './linter/used_module_finder'
-path = require 'path'
-
-
 class Linter
 
 
-  constructor: (@dir, {allowUnused, devFilePatterns, devScripts, ignoreFilePatterns}) ->
-    @dependencyLinter = new DependencyLinter {allowUnused, devFilePatterns, devScripts}
-    @usedModuleFinder = new UsedModuleFinder {@dir, ignoreFilePatterns}
+  constructor: (@dir, {@allowUnused, @devFilePatterns, @devScripts, @ignoreFilePatterns}) ->
 
 
-  lint: (done) ->
-    async.parallel {
-      listedModules: @getListedModules
-      usedModules: @usedModuleFinder.find
-    }, asyncHandlers.transform(@dependencyLinter.lint, done)
+  lint: (done) =>
+    async = require 'async'
+    asyncHandlers = require 'async-handlers'
```

Comment:
Re point 1: the complexity increases because the method now has an additional assignment and method call.